include/oneapi/tbb/detail/_flow_graph_impl.h

```diff
@@ -359,21 +361,23 @@ class graph : no_copy, public graph_proxy {
 
 };  // class graph
 
+template<typename DerivedType>
 inline void graph_task::destruct_and_deallocate(const execution_data& ed) {
     auto allocator = my_allocator;
     // TODO: investigate if direct call of derived destructor gives any benefits.
     this->~graph_task();
-    allocator.deallocate(this, ed);
+    allocator.deallocate(static_cast<DerivedType*>(this), ed);
 }
 
+template<typename DerivedType>
 inline void graph_task::finalize(const execution_data& ed) {
     graph& g = my_graph;
-    destruct_and_deallocate(ed);
+    destruct_and_deallocate<DerivedType>(ed);
     g.release_wait();
 }
 
 inline task* graph_task::cancel(execution_data& ed) {
-    finalize(ed);
+    finalize<graph_task>(ed);
     return nullptr;
 }
```

Comment:
Do we really need to override `cancel` in the `graph_task` now?

Reply:
Removed